### PR TITLE
Updated dockerfile to optimize layers and caching of pip and OS packa…

### PR DESCRIPTION
**Pull request by jrgwv**
*Issue:*
Investigate optimization areas for dockerfile with hadolint utility. 

*Description of changes:*
Ran dockerfile through hadolint to identify improvement areas and bedrock to optimize. As a result the Image size reduced 16.2 MiB Blob size 13 MiB.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,37 +2,34 @@
...
 
-# Install system dependencies
-RUN apt-get update && apt-get install -y build-essential
-
-# Install Python dependencies
+# Install system dependencies and Python dependencies
 COPY ./server/requirements.txt /server/
-RUN pip wheel --no-cache-dir --no-deps --wheel-dir /server/wheels -r requirements.txt
+RUN apt-get update && apt-get install -y build-essential --no-install-recommends \
+    && apt-get clean \
```
Comment (JoeNorth) [inline Dockerfile:14]: Cleanup here is unnecessary as this is just in the builder stage

File: Dockerfile [hunk 2]
```diff
@@ -2,37 +2,34 @@
...
+    && rm -rf /var/lib/apt/lists/* \
```
Comment (JoeNorth) [inline Dockerfile:15]: Same as above

File: Dockerfile [hunk 3]
```diff
@@ -2,37 +2,34 @@
...
-# Install system dependencies
-RUN apt-get update && apt-get install -y netcat
-
-# Install Python dependencies
+# Install system dependencies and Python dependencies
 COPY --from=builder /server/wheels /server/wheels
 COPY --from=builder /server/requirements.txt .
-RUN pip install --no-cache /server/wheels/*
-RUN pip install uvicorn
+RUN apt-get update && apt-get install -y netcat --no-install-recommends \
```
Comment (JoeNorth) [inline Dockerfile:25]: Is it necessary to install netcat? I don't think anything has a dependency on it.

File: Dockerfile [hunk 4]
```diff
@@ -2,37 +2,34 @@
...
+    && pip install --no-cache-dir uvicorn
 
 # Copy project
 COPY . /server/
 
 # Expose the port the app runs in
 EXPOSE 8000
-
-# Define the command to start the container
-CMD uvicorn server.app.main:app --host 0.0.0.0 --port 8000
```
Comment (JoeNorth) [inline Dockerfile:38]: Why is this being removed? This is needed to start the server.

**Review by JoeNorth — COMMENTED** [review]
A few comments to consider before moving forward.

File: Dockerfile (hunk 3 above)
Comment (jrgwv) [reply Dockerfile:25]: Probably not I kept this from the original dockerfile, will remove.

File: Dockerfile (hunk 2 above)
Comment (jrgwv) [reply Dockerfile:15]: Carryover form original file, will remove and test.

**Comment by jrgwv** [comment]
Updated based on feedback and updated the requirements.txt to address two high security issues that had been open a while. Tested locally.

**Review by JoeNorth — APPROVED** [review]
LGTM

**Pull request merged**